darts/models/forecasting/query_selector.py

```diff
@@ -0,0 +1,650 @@
+            dim_val, dim_val * 4, dropout=0.1
+        )
+        self.feed_forward_network_dec = self.feed_forward_network_dec(
+            dim_val, dim_val * 4, dropout=0.1
+        )
+
+        # Initiate encoder and Decoder layers
+        self.encs = []
+        for i in range(n_encoder_layers):
+            self.encs.append(
+                EncoderLayer(
+                    dim_val,
+                    dim_attn,
+                    feed_forward=self.feed_forward_network_enc,
+                    n_heads=n_heads,
+                    attn_type=enc_attn_type,
+                    norm=self.layer_norm,
+                )
+            )
+        self.encs = InferenceModuleList(self.encs)
+        self.decs = []
+        for i in range(n_decoder_layers):
+            self.decs.append(
+                DecoderLayer(
+                    dim_val,
+                    dim_attn,
+                    feed_forward=self.feed_forward_network_dec,
+                    n_heads=n_heads,
+                    attn_type=dec_attn_type,
+                    norm=self.layer_norm,
+                )
+            )
+        self.decs = InferenceModuleList(self.decs)
+        self.pos = PositionalEncoding(dim_val)
+
+        self.enc_dropout = MonteCarloDropout(p=dropout)
+
+        # Dense layers for managing network inputs and outputs
+        self.enc_input_fc = nn.Linear(self.input_dim, dim_val)
+        self.dec_input_fc = nn.Linear(self.input_dim, dim_val)
+        self.out_fc = nn.Linear(
+            dec_seq_len * dim_val,
+            self.output_chunk_length * self.output_dim * self.nr_params,
+        )
+
+        self.debug = debug
+
+    def forward(self, x_in: tuple):
+        x, _ = x_in
+        # x: [Batch, Input length, Channel]
+
+        # encoder
+        e = self.encs[0](self.pos(self.enc_dropout(self.enc_input_fc(x))))
+
+        for enc in self.encs[1:]:
+            e = enc(e)
+        if self.debug:
+            print(f"Encoder output size: {e.shape}")
+        # decoder
+        decoded = self.dec_input_fc(x[:, -self.dec_seq_len :])
+
+        d = self.decs[0](decoded, e)
+        for dec in self.decs[1:]:
+            d = dec(d, e)
+
+        # output
+        x = self.out_fc(d.flatten(start_dim=1))
+
+        x = torch.reshape(x, (x.shape[0], -1, self.output_dim))
+        x = x.view(
+            x.shape[0], self.output_chunk_length, self.output_dim, self.nr_params
+        )
+        return x
+
+    def record(self):
+        self.debug = True
+        for enc in self.encs:
+            enc.record()
+        for dec in self.decs:
+            dec.record()
+
+
+class QuerySelectorModel(PastCovariatesTorchModel):
+    def __init__(
+        self,
+        input_chunk_length: int,
+        output_chunk_length: int,
+        dim_val: int = 32,
+        dim_attn: int = 128,
+        dec_seq_len: int = 12,
+        n_encoder_layers: int = 3,
+        n_decoder_layers: int = 3,
+        enc_attn_type="query_selector_0.8",
```

Comment:
Could we keep the fraction f of the query selector as a separate float argument?